src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,146 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserInvalidDataException;
+import core.basesyntax.model.User;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class RegistrationServiceImplTest {
+    private static final int EXPECTED_ID = 1;
+    private static final int MIN_CORRECT_AGE = 18;
+    private static final int MAX_INCORRECT_AGE = 0;
+    private static final int MIN_INCORRECT_AGE = -100;
+    private static final int MIN_AGE_LESS_18 = 1;
+    private static final int MAX_CORRECT_TESTING_AGE = 100;
+    private static final int METHOD_EXECUTION_SEQUENCE_NUMBER = 1;
+    private static final String CORRECT_LOGIN = "correctLogin";
+    private static final String TEST_LOGIN_SECOND = "testLoginSecond";
+    private static final String TEST_LOGIN_FOR_AGE = "testLoginForAge";
+    private static final Map<String, String> passwordsByLogin = Map.of(
+            "correctLogin", "correctPassword",
+            "testLoginSecond", "testPasswordSecond"
+    );
+    private static final List<String> shortLogins = List.of(
+            "", "t", "tt", "ttt", "tttt", "ttttt");
+    private static final List<String> shortPasswords = List.of(
+            "", "p", "pp", "ppp", "pppp", "ppppp");
+
+    private static StorageDao storageDao;
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void initVariables() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUser() {
+        user = new User();
+        user.setLogin(CORRECT_LOGIN);
+        user.setPassword(passwordsByLogin.get(CORRECT_LOGIN));
+        user.setAge(MIN_CORRECT_AGE);
+    }
+
+    @Order(METHOD_EXECUTION_SEQUENCE_NUMBER)
+    @Test
+    void register_validUser_Ok() {
+        User actualUser = registrationService.register(user);
+
+        assertEquals(user, actualUser);
+        assertEquals(EXPECTED_ID, actualUser.getId());
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        runAssertThrows(null);
+    }
+
+    @Test
+    void register_loginPresentInStorage_notOk() {
+        user.setLogin(TEST_LOGIN_SECOND);
+        user.setPassword(passwordsByLogin.get(TEST_LOGIN_SECOND));
+        storageDao.add(user);
+
+        runAssertThrows(user);

```

Comment:
1. You should try to register different users with same login
2. 
![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/6caa8a13-717f-40b2-be01-3b05c8705178)